src/ripple/ripple.js

```diff
@@ -118,7 +119,12 @@ MaterialRipple.prototype.upHandler_ = function(event) {
 
   // Don't fire for the artificial "mouseup" generated by a double-click.
   if (event && event.detail !== 2) {
-    this.rippleElement_.classList.remove(this.CssClasses_.IS_VISIBLE);
+    // Allow a repaint to occur before removing this class, so the animation
+    // shows for tap events, which seem to trigger a mouseup too soon after
+    // mousedown.
+    window.setTimeout(function() {
```

Comment:
As I was walking around this evening I had somewhat of a revelation.

60FPS === a tick every 16.666.... MS ... so 17 becomes way less magical if react was optimizing here for a 60FPS system (where at most we could paint 60 frames a secong).

Given that knowledge, I would argue that TINY_TIMEOUT should actually be `17` 

:cool: 

![](http://ak-hdl.buzzfed.com/static/2015-02/1/20/enhanced/webdr08/enhanced-32005-1422840611-40.jpg)